strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassFileIndex.java

```diff
@@ -114,66 +112,59 @@ private void replaceVertex(JavaElement newClass) {
 
     private void updateJavaElementWithImportReferences(JavaElement javaElementParentClass,
         String[] imports) {
+
         for (String importz : imports) {
+
+            if (addImport(javaElementParentClass, importz)) {
+                if (filter.shouldBeIncluded(importz)) {
+                    JavaClass javaClass = builder.getClass(importz);
+                    if (javaClass != null) {
+                        updateJavaElementWithImportReferences(javaElementParentClass, javaClass.getImports());
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean addImport(JavaElement javaElementParentClass, String importz) {
+
+        if (!importz.startsWith("java")) {
```

Comment:
This is because of nature of recursive thing, if you want we can discuss in bjn and I explain why I did in this way, it is just a matter of not adding information that it is not useful